File: extract_images.js
const fs = require('fs');
const text = fs.readFileSync('prado.html', 'utf8');
const regex = /https:\/\/[^"'>\s]+\.(?:jpg|jpeg)/g;
const urls = text.match(regex) || [];
const carwowUrls = [];
const fallbackUrls = [];
for (const url of urls) {
  if (!fallbackUrls.includes(url)) fallbackUrls.push(url);
  if (
    (url.includes('carwow') || url.includes('caranddriver') || url.includes('hips.hearstapps.com')) &&
    !url.includes('headshots') &&
    !url.includes('wheel')
  ) {
    if (!carwowUrls.includes(url)) {
      carwowUrls.push(url);
      if (carwowUrls.length >= 5) break;
    }
  }
}
const output = carwowUrls.length ? carwowUrls : fallbackUrls.slice(0, 20);
for (const url of output) {
  console.log(url);
}

File: extract_images.py
import re
from pathlib import Path
text = Path('prado.html').read_text(errors='ignore')
urls = re.findall(r'https://[^"\'\>\s]+\.(?:jpg|jpeg|png)', text)
seen = []
for url in urls:
    if 'carwow' in url or 'caranddriver' in url or 'hips.hearstapps.com' in url:
        if url not in seen:
            seen.append(url)
            print(url)
            if len(seen) >= 10:
                break
